homeassistant/components/tesla/device_tracker.py

```diff
@@ -1,45 +1,70 @@
 """Support for tracking Tesla cars."""
 import logging
 
-from homeassistant.helpers.event import async_track_utc_time_change
-from homeassistant.util import slugify
+from homeassistant.components.device_tracker import SOURCE_TYPE_GPS
+from homeassistant.components.device_tracker.config_entry import TrackerEntity
 
-from . import DOMAIN as TESLA_DOMAIN
+from . import DOMAIN as TESLA_DOMAIN, TeslaDevice
 
 _LOGGER = logging.getLogger(__name__)
 
 
-async def async_setup_scanner(hass, config, async_see, discovery_info=None):
-    """Set up the Tesla tracker."""
-    tracker = TeslaDeviceTracker(
-        hass, config, async_see, hass.data[TESLA_DOMAIN]["devices"]["devices_tracker"]
-    )
-    await tracker.update_info()
-    async_track_utc_time_change(hass, tracker.update_info, second=range(0, 60, 30))
-    return True
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Tesla binary_sensors by config_entry."""
+    entities = [
+        TeslaDeviceEntity(
+            device,
+            hass.data[TESLA_DOMAIN][config_entry.entry_id]["controller"],
+            config_entry,
+        )
+        for device in hass.data[TESLA_DOMAIN][config_entry.entry_id]["devices"][
+            "devices_tracker"
+        ]
+    ]
+    async_add_entities(entities, True)
 
 
-class TeslaDeviceTracker:
+class TeslaDeviceEntity(TrackerEntity, TeslaDevice):
```

Comment:
It's more future proof to invert the inheritance order and start with the `TeslaDevice`, since we're relying on the init method of that class.